process_duty_email.py

```diff
@@ -0,0 +1,293 @@
+"""
+Process duty emails
+Developer: Igor Malashchuk 
+Email: igor.malashchuk@nhs.net
+Date Modified: 14/09/2022
+"""
+import subprocess
+import os
+import re
+import collections
+from datetime import datetime
+import pandas
+import math
+import numpy as np
+import config
+
+# Run script in Powershell: duty
+# run script in command prompt: S:\Genetics_Data2\Array\Software\Python-3.6.5\python S:\Genetics_Data2\Array\Software\duty_bioinformatician_scripts\process_duty_email.py
+version = "2.0.0"
+
+def ask_for_folder():
+    """
+    For MokaPipe the destination folder can be different.
+    Therefore this function ask the user to enter the last part of the folder
+    """
+    output_folder = ""
+    path_worksheets = config.MokaPipe_path_worksheets
+    print("Part of destination is:{}".format(path_worksheets))
+    while len(output_folder) == 0:
+        output_folder = input(
+            'Please complete the final part of the destination folder \n(e.g."NGS_401 to 500\\NGS484")\nMay need to place inside double quotes in Powershell:'
+        )
+        destination_folder = path_worksheets + output_folder
+        print("The destination folder is : {}".format(destination_folder))
+        if destination_folder == path_worksheets or not os.path.isdir(destination_folder):
+            print("The directory does not exist. Please try again!")
+            output_folder = ""
+    return path_worksheets + output_folder
+
+def get_data(df, path_to_folder):
+    """
+    This function reads the data frame (df)..
+    From the df file it obtains the download links and creates a string argument.   
+    """
+    all_urls = ""
+    for index, row in df.iterrows():
+        line = row['url'] + "," + path_to_folder + "£$%"

```

Comment:
£$%?

Reply:
i tried to make the separator as unique as possible so that I could separate only at the correct location. I created it as a global variable